cai_causal_graph/graph_components.py

```diff
@@ -110,6 +169,44 @@ def variable_type(self, new_type: Union[NodeVariableType, str]):
             raise TypeError(f'Expected NodeVariableType or string, got object of type {type(new_type)}.')
         self._variable_type = new_type
 
+    def _check_var_name_is_valid(self, name: str):
+        """Check that the name is valid."""
+        vname, _ = get_variable_name_and_lag(name)
+        # do the same for the current node
+        vname_this, _ = get_variable_name_and_lag(self.identifier)
+
+        # check that the name is valid relative to the given Node
+        assert vname_this == vname, f'Invalid node name: {name}.'
+
+    @property
+    def variable_name(self) -> Optional[str]:
+        """Return the variable name of the node from the metadata."""
+        return self.meta.get('variable_name', None)
+
+    @variable_name.setter
+    def variable_name(self, new_name: str):
+        """
+        Set the variable name of the node.
+
+        :param new_name: New variable name.
+        """
+        self._check_var_name_is_valid(new_name)
+        self.meta['variable_name'] = new_name
+
+    @property
+    def time_lag(self) -> int:
+        """Return the time lag of the node from the metadata."""
+        return self.meta.get('time_lag', 0)
+
+    @time_lag.setter
+    def time_lag(self, new_lag: int):
+        """
+        Set the time lag of the node.
+
+        :param new_lag: New time lag.
+        """
+        self.meta['time_lag'] = new_lag
```

Comment:
Ah yeah just commented the same things. Think this should all be in a subclass of Node.

Reply:
Then tbh I don't think we'd even need to use meta, could just be properties of the class

Reply:
> Then tbh I don't think we'd even need to use meta, could just be properties of the class

I would be very against this! It will lead to serialization issues in the future potentially (same issues we have between `causal_graph.CausalGraph` and `cai_causal_graph.CausalGraph`). I think it should be kept in `meta` with the getter and setter as is done right now - but just on a subclass.

Reply:
I moved the code in the TimeSeriesNode. I don't think the setter should be there as a node cannot change lag unless it's replaced. Regarding the meta/properties arguments, I just followed the specs.